### PR TITLE
[FAB-17357] Add endorser metric for simulation failure

### PR DESCRIPTION
**Pull request by stephyee**
Signed-off-by: Tiffany Harris <tiffany.harris@ibm.com>

#### Type of change

- Bug fix
- Documentation update

#### Description

- Add new endorser metric, _endorser_simulation_failures_
- Add unit test for endorser_simulation_failures metric

#### Related issues
[FAB-17357](https://jira.hyperledger.org/browse/FAB-17357)

### CODE REVIEW

**Comment by joealewine** [comment]
Doc change lgtm

**Review by denali49 — APPROVED** [review]
Doc change approved.

**Comment by yacovm** [comment]
> Doc change approved.

Are you aware that this is a [machine-generated](https://github.com/hyperledger/fabric/blob/master/scripts/metrics_doc.sh) documentation change?

**Review by joealewine — APPROVED** [review]

File: core/endorser/endorser.go [hunk 1]
```diff
@@ -111,8 +111,14 @@ func (e *Endorser) callChaincode(txParams *ccprovider.TransactionParams, input *
 		logger.Infof("finished chaincode: %s duration: %dms", chaincodeName, elapsedMillisec)
 	}(time.Now())
 
+	meterLabels := []string{
+		"channel", txParams.ChannelID,
+		"chaincode", chaincodeName,
```
Comment (yacovm) [inline core/endorser/endorser.go:116]: Are we sure we want to have this chaincode as well ? I thought that the general rule of thumb is not have something in the labels that has a large domain? 

pinging @sykesm , wdyt?

**Comment by denali49** [comment]
FAB-17358 was associated with this and requested a doc review to correct a
typo in the .rst file.

**Comment by denali49** [comment]
Meant to say FAB-15578 was related to typo in the .rst file.  Please disregard original response.
Thanks
Chris

File: core/endorser/endorser.go (hunk 1 above)
Comment (wlahti) [reply core/endorser/endorser.go:116]: Seems consistent with other metrics in this file.

Comment (yacovm) [reply core/endorser/endorser.go:116]: Indeed!

**Review by yacovm — APPROVED** [review]

**Comment by denyeart** [comment]
Unfortunately, this caused a regression - FAB-17525.

**Pull request merged**